bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,433 @@
+    private @Nullable InetAddress ipAddress = null;
+    public @Nullable CGateSession cGateSession = null;
+    private @Nullable ScheduledFuture<?> keepAliveFuture = null;
+
+    public CBusCGateHandler(Bridge br) {
+        super(br);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Nothing to do here
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.OFFLINE);
+        logger.debug("Initializing CGate Bridge handler.");
+        Optional<CBusCGateConfiguration> cfg = Optional.of(getConfigAs(CBusCGateConfiguration.class));
+        if (cfg.isPresent()) {
+            configuration = cfg.get();
+            logger.debug("Using configuration {}", configuration);
+            if ("127.0.0.1".equals(configuration.ipAddress) || "localhost".equals(configuration.ipAddress)) {
+                this.ipAddress = InetAddress.getLoopbackAddress();
+            } else {
+                try {
+                    this.ipAddress = InetAddress.getByName(configuration.ipAddress);
+                } catch (UnknownHostException e1) {
+                    updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                            "IP Address not resolvable");
+                    return;
+                }
+            }
+            InetAddress address = this.ipAddress;
+            if (address != null)
+                logger.debug("CGate IP         {}.", address.getHostAddress());
+
+            keepAliveFuture = scheduler.scheduleWithFixedDelay(() -> {
+                keepAlive();
+            }, 0, 100, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR);
+        }
+    }
+
+    private void keepAlive() {
+        CGateSession session = cGateSession;
+        if (session == null || !session.isConnected()) {
+            if (!getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                connect();
+            } else {
+                updateStatus();
+            }
+        }
+    }
+
+    private void connect() {
+        CGateSession cGateSession = this.cGateSession;
+        if (cGateSession == null) {
+            cGateSession = CGateInterface.connect(this.ipAddress, 20023, 20024, 20025, new CBusThreadPool());
+            cGateSession.registerEventCallback(new EventMonitor());
+            cGateSession.registerStatusChangeCallback(new StatusChangeMonitor());
+            this.cGateSession = cGateSession;
+        }
+        if (cGateSession.isConnected()) {
+            logger.debug("CGate session reports online");
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            try {
+                cGateSession.connect();
+                updateStatus();
+            } catch (CGateConnectException e) {
+                updateStatus();
+                logger.warn("Failed to connect to CGate: {}", e.getMessage());
```

Comment:
The user should already have been notified by the things status changing to offline. This doesn't need to be logged as a warning, change to debug.

Reply:
Fixed